src/mscorlib/src/System/Globalization/CompareInfo.cs

```diff
@@ -802,6 +801,53 @@ public unsafe virtual int IndexOf(string source, string value, int startIndex, i
             return IndexOfCore(source, value, startIndex, count, options, null);
         }
 
+        // The following IndexOf overload is mainly used by String.Replace. This overload assume the parameters are already validated
+        // and the caller is passing a valid matchLengthPtr pointer.
+        internal unsafe int IndexOf(string source, string value, int startIndex, int count, CompareOptions options, int* matchLengthPtr)
```

Comment:
It seems that matchLengthPtr is just a way of passing the length of the match (which is always either null or value.Length, unless you end up calling FindNLSStringEx). It's only a pointer because that's what FindNLSStringEx requires. So why not replace it everywhere in the managed code with just a ref int ?